### PR TITLE
DER-87 - Eliminate notions related to transactions in the various lower level derivatives ontologies

### PR DESCRIPTION
**Pull request by ElisaKendall**
Signed-off-by: Elisa Kendall <ekendall@thematix.com>

## Description

This resolution includes merging a number of derivatives ontologies and concepts that distinguished between derivative transactions and contracts unnecessarily, and that distinguished between over-the-counter and listed derivatives, which added confusion and unnecessary duplication.  It also resolves several circular references between the remaining ontologies as a consequence.

Fixes: #1237 / DER-87

## Checklist:

- [x] I'm familiar with the [FIBO developer quide](../CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [x] My contribution follows the [principles of best practices for FIBO](../ONTOLOGY_GUIDE.md).
- [x] My changes have been reconciled with latest master and no merge conflicts remain.
- [x] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [x] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [x] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

File: DER/FxDerivatives/FxOptions.rdf [hunk 1]
```diff
@@ -64,193 +68,109 @@
...
-		<rdfs:label xml:lang="en">fx option premium</rdfs:label>
-		<rdfs:domain rdf:resource="&fibo-der-fx-opt;ForeignExchangeOption"/>
-		<rdfs:range rdf:resource="&fibo-fnd-acc-cur;MonetaryAmount"/>
-		<skos:definition xml:lang="en">FpML &apos;Premium amount or premium installment amount for an option.&apos;</skos:definition>
-	</owl:ObjectProperty>
-	
-	<owl:Class rdf:about="&fibo-der-fx-opt;ForeignExchangeOptionSeller">
-		<rdfs:label xml:lang="en">fx option seller</rdfs:label>
+		<rdfs:label xml:lang="en">currency option</rdfs:label>
+		<skos:definition xml:lang="en">A contract that grants the holder the right, but not the obligation, to buy or sell currency at a specified exchange rate during a specified period of time.</skos:definition>
```
Comment (rivettp) [inline DER/FxDerivatives/FxOptions.rdf:94]: important - need to say that it's a specific amount of the currency. Need a restriction for the amount. Is the contract in terms of the buying or the selling currency e.g. do you say I want to buy $100 of GBP or 100 GBP?

File: DER/FxDerivatives/FxOptions.rdf [hunk 2]
```diff
@@ -64,193 +68,109 @@
...
-		<rdfs:label xml:lang="en">fx call option contract</rdfs:label>
+	<owl:Class rdf:about="&fibo-der-fx-opt;ForeignExchangeOptionLeg">
+		<rdfs:subClassOf rdf:resource="&fibo-der-fx-ctr;ForeignExchangeDerivative"/>
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-fi-fi;Option"/>
+		<rdfs:label xml:lang="en">foreign exchange option leg</rdfs:label>
+		<skos:definition xml:lang="en">A standard FX OTC option (European or American) which may be a complete trade in its own right or part of a trade strategy. FpML A type that is used for describing a standard FX OTC option (European or American) which may be a complete trade in its own right or part of a trade strategy.</skos:definition>
 	</owl:Class>
 	
-	<owl:Class rdf:about="&fibo-der-fx-opt;FxPutOptionContract">
+	<owl:Class rdf:about="&fibo-der-fx-opt;ForeignExchangePutOption">
```
Comment (rivettp) [inline DER/FxDerivatives/FxOptions.rdf:131]: naming: when do we use Fx and when ForeignExchange?

File: DER/FxDerivatives/FxOptions.rdf [hunk 3]
```diff
@@ -64,193 +68,109 @@
...
-	<owl:ObjectProperty rdf:about="&fibo-der-fx-opt;hasCorrespondingContract">
-		<rdfs:label xml:lang="en">has corresponding contract</rdfs:label>
-		<rdfs:domain rdf:resource="&fibo-der-fx-opt;ForeignExchangeOptionTransaction"/>
-		<rdfs:range rdf:resource="&fibo-der-fx-opt;ForeignExchangeOption"/>
+	<owl:ObjectProperty rdf:about="&fibo-der-fx-opt;hasBuyingCurrency">
+		<rdfs:subPropertyOf rdf:resource="&fibo-fnd-acc-cur;hasCurrency"/>
+		<rdfs:label xml:lang="en">has buying currency</rdfs:label>
+		<rdfs:domain rdf:resource="&fibo-der-fx-opt;CurrencyOption"/>
+		<rdfs:range rdf:resource="&fibo-fnd-acc-cur;Currency"/>
+		<skos:definition xml:lang="en">The currency in which the option is bought.</skos:definition>
```
Comment (rivettp) [inline DER/FxDerivatives/FxOptions.rdf:161]: ambiguous: is it for buying the option or buying the cash when the option is exercised? I could buy an option in USD that gives me the future right to buy GBP in EUR

File: DER/FxDerivatives/FxContracts.rdf [hunk 4]
```diff
@@ -30,42 +54,162 @@
...
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&lcc-cr;hasPart"/>
+				<owl:someValuesFrom rdf:resource="&fibo-der-fx-ctr;ForeignExchangeSwap"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">foreign exchange forward outright</rdfs:label>
+		<skos:definition xml:lang="en">forward contract in the forex market that locks in the price at which an entity can buy or sell a currency on a future date. Also known as &quot;outright forward currency transaction&quot;, &quot;forward outright&quot; or &quot;FX forward&quot;.</skos:definition>
```
Comment (rivettp) [inline DER/FxDerivatives/FxContracts.rdf:110]: No, "also known as..." must be a modeled as a synonym.

File: DER/FxDerivatives/FxContracts.rdf [hunk 5]
```diff
@@ -30,42 +54,162 @@
...
+		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">In currency forward contracts, the contract holders are obligated to buy or sell the currency at a specified price, at a specified quantity and on a specified future date. These contracts cannot be transferred. Jan 10 Review Notes Outright Forward is the term for the professional markets. Spot + Swap where Swap is 2 simultaneous transactions.</fibo-fnd-utl-av:explanatoryNote>
```
Comment (rivettp) [inline DER/FxDerivatives/FxContracts.rdf:111]: use singular "in A currency forward contract the contract holder IS..."

File: DER/FxDerivatives/FxContracts.rdf [hunk 6]
```diff
@@ -30,42 +54,162 @@
...
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-agr-ctr;hasContractualElement"/>
+				<owl:someValuesFrom rdf:resource="&fibo-der-der-sp;SpotCashSettlementTerms"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">foreign exchange spot contract</rdfs:label>
+		<skos:definition xml:lang="en">The rate of a foreign-exchange contract for immediate delivery. Also known as &quot;benchmark rates&quot;, &quot;straightforward rates&quot; or &quot;outright rates&quot;, spot rates represent the price that a buyer expects to pay for a foreign currency in another currency.</skos:definition>
```
Comment (rivettp) [inline DER/FxDerivatives/FxContracts.rdf:130]: No the definition is about Rates but the class is for a Contract.

File: DER/FxDerivatives/FxContracts.rdf [hunk 7]
```diff
@@ -30,42 +54,162 @@
...
+		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">Though the spot exchange rate is said to be settled immediately, the globally accepted settlement cycle for foreign-exchange contracts is two days. Foreign-exchange contracts are therefore settled on the second day after the day the deal is made.</fibo-fnd-utl-av:explanatoryNote>
```
Comment (rivettp) [inline DER/FxDerivatives/FxContracts.rdf:131]: "second day after" is not very clear. so it's T+2? I assume that's business days?

File: DER/ExchangeTradedDerivatives/ExchangeTradedOptions.rdf [hunk 8]
```diff
@@ -98,54 +103,50 @@
...
-	</owl:Class>
-	
-	<owl:Class rdf:about="&fibo-der-etd-eto;BondOptionUnderlying">
-		<rdfs:subClassOf rdf:resource="&fibo-der-etd-eto;OptionOnFuturesUnderlying"/>
-		<rdfs:label xml:lang="en">bond option underlying</rdfs:label>
-		<skos:definition xml:lang="en">The underlying on a Bond Option, which is the corresponding Bond Future contract on the same exchange.</skos:definition>
-		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">This is the underlying in the sense that when the option is exercised the investor takes delivery of the corresponding futures contract.</fibo-fnd-utl-av:explanatoryNote>
+		<skos:definition xml:lang="en">option on a futures contract for a bond</skos:definition>
+		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">When exercised, the holder exercises the right to buy the future on a Bond, s/he does not exercise the right to buy a bond. For example an option on a Treasury Bill is an option on a T Bill Future, not on a T Bill itself. Hence these are defined as a kind of futures option.</fibo-fnd-utl-av:explanatoryNote>
```
Comment (rivettp) [inline DER/ExchangeTradedDerivatives/ExchangeTradedOptions.rdf:134]: not very clear. And I think "a" in the phase "option on a Treasury Bill" seems wrong.

**Review by rivettp — APPROVED** [review]
I made some significant comments which should be addressed but I'm OK for them to be done in the next pass.
We really need some examples next time.

**Review by trypuz — APPROVED** [review]
Approved

**Pull request merged**